doc/influxql-2.md

```diff
@@ -0,0 +1,910 @@
+
+The join expression can also pull in values from another matrix. In that case the series that gets joined will be based on the `keys` passed in. For example:
+
+```javascript
+var e = select(database:"foo",where:{'metric'="errors"})
+	      .range(start:-24h).window(period:10m).sum()
+var r = select(database:"foo",where:{'metric'="requests"})
+          .range(start:-24h).window(period:10m).sum()
+
+e.join(group:["host"],exp:{
+	('metric'."errors".float()/r.'metric'."requests").float(fixed:2) AS "error_rate"})
+```
+
+In the above example we see `r.'metric'."requests"`. When join goes through and joins series in `e` it will match against the series in `r` that have matching `host` tags.
+
+#### Union
+`union` will append all the series in one matrix to another.
+
+```javascript
+// Accepts: matrix
+// Arguments:
+//   m - required matrix to union
+// Returns: matrix
+// for example, merge events stream from two databases
+select(database:"foo",where:{'metric' = "events"})
+  .range(start:-1h)
+  .union(m:select(database:"bar",where:{'metric' = "events"})
+    .range(start:-1h))
+```
+
+#### Filter
+`filter` will filter points out of a matrix based on an expression. It can be used to filter points from one series based on evaluations from another series.
+
+```javascript
+// Accepts: matrix
+// Arguments:
+//   exp - a required expression string
+//   group - a string array of keys to group series by
+// Returns: matrix
+select(database:"foo",where:{'metric'="cpu_user"})
+  .range(start:-1h)
+  .filter(exp:{$ < 50}) // filter out any values less than 50
+
+// or to filter series in a matrix based on others
+select(database:"foo",where:{
+    'metric'="response_time" OR 'metric'="log_lines"})
+  .filter(group: ["host"], exp:{'metric'."response_time" > 300})
+```
+
+#### Sort
+`sort` will sort the series in a matrix in ascending order by default. They can either be sorted by their tags or by some expression.
+
+```javascript
+// Accepts: matrix
+// Arguments:
+//   exp - an expression to specify what to sort by. Should return a value that can be sorted
+//   keys - a string array of tag keys to specify the sort order, required if exp not given
+//   order - string of either asc or desc, asc is default
+// Returns: the original matrix, modified in place with the sort order
+select(database:"foo",where:{'metric'="cpu_system"})
+  .range(start:-1h)
+  .sort(exp:{mean($)})
+  .limit(n:10)
+```
+
+#### Rate
+`rate` calculates the average rate of increase between each point for each series in a matrix.
+
+```javascript
+// Accepts: a matrix
+// Arguments:
+//   precision - a duration for what the rate should be. per second, per minute, etc.
```

Comment:
Should this parameter be named `unit` instead of precision? Its talking about the denominator of the rate calculation not necessarily the precision of the calculation.